otelcol/config.go

```diff
@@ -38,6 +38,9 @@ type Config struct {
 	// Processors is a map of ComponentID to Processors.
 	Processors map[component.ID]component.Config
 
+	// Connectors is a map of ComponentID to connectors.
+	Connectors map[component.ID]component.Config
```

Comment:
Let's add the "featuregate" for the connectors now, and in the Config.Validate fail if len(Connectors) != 0 and the gate is disable. The reason is because I want to make sure we fail fast and people don't try to use connectors if they are not fully implemented/enabled.

Reply:
I've added the feature gate, and a check in the validate func.